zeebe/engine/src/main/java/io/camunda/zeebe/engine/processing/job/JobUpdateProcessor.java

```diff
@@ -0,0 +1,73 @@
+/*
+ * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH under
+ * one or more contributor license agreements. See the NOTICE file distributed
+ * with this work for additional information regarding copyright ownership.
+ * Licensed under the Camunda License 1.0. You may not use this file
+ * except in compliance with the Camunda License 1.0.
+ */
+package io.camunda.zeebe.engine.processing.job;
+
+import io.camunda.zeebe.engine.processing.streamprocessor.TypedRecordProcessor;
+import io.camunda.zeebe.engine.processing.streamprocessor.writers.StateWriter;
+import io.camunda.zeebe.engine.processing.streamprocessor.writers.TypedRejectionWriter;
+import io.camunda.zeebe.engine.processing.streamprocessor.writers.TypedResponseWriter;
+import io.camunda.zeebe.engine.processing.streamprocessor.writers.Writers;
+import io.camunda.zeebe.engine.state.immutable.JobState;
+import io.camunda.zeebe.engine.state.immutable.ProcessingState;
+import io.camunda.zeebe.protocol.impl.record.value.job.JobRecord;
+import io.camunda.zeebe.protocol.record.RejectionType;
+import io.camunda.zeebe.protocol.record.intent.JobIntent;
+import io.camunda.zeebe.scheduler.clock.ActorClock;
+import io.camunda.zeebe.stream.api.records.TypedRecord;
+
+public class JobUpdateProcessor implements TypedRecordProcessor<JobRecord> {
+
+  private static final String NO_JOB_FOUND_MESSAGE =
+      "Expected to update job with key '%d', but no such job was found";
+  private static final String NEGATIVE_RETRIES_MESSAGE =
+      "Expected to update job with key '%d' with a positive amount of retries, "
+          + "but the amount given was '%d'";
+
+  private final JobState jobState;
+  private final StateWriter stateWriter;
+  private final TypedRejectionWriter rejectionWriter;
+  private final TypedResponseWriter responseWriter;
+
+  public JobUpdateProcessor(final ProcessingState state, final Writers writers) {
+    jobState = state.getJobState();
+    stateWriter = writers.state();
+    rejectionWriter = writers.rejection();
+    responseWriter = writers.response();
+  }
+
+  @Override
+  public void processRecord(final TypedRecord<JobRecord> command) {
```

Comment:
I didn't do that in this PR because the 2 processor are implemented in a different way. `JobUpdateTimeoutProcessor ` implements `TypedRecordProcessor` while `JobUpdateRetriesProcessor` implements `CommandProcessor` (tech debt)

I already planned to refactor the job related processors as a follow-up issue 👍

Reply:
Yeah, I saw that as well. Nonetheless, we should be able to extract the base functionality of retries and timeout updates into common utility methods we can call from the three involved processors, right? That way, thei

Reply:
Yes, that's the plan

Reply:
OK, that suffices for me, just wanted to make sure we have that on our radar and are aware that we should align those three processors so they'll stay in sync whenever we change something to retries or timeout update 👍